test/LocatedTriplesTest.cpp

```diff
@@ -171,3 +171,197 @@ TEST_F(LocatedTriplesTest, mergeTriples) {
     locatedTriplesPerBlock.mergeTriples(2, std::nullopt, result, 0, V(1));
   }
 }
+
+// Test the locating of triples in a permutation using `locatedTriple`.
+TEST_F(LocatedTriplesTest, locatedTriple) {
+  // The actual test, for a given block size.
+  //
+  // TODO: Also make the permutation an argument, right now it's only PSO.
+  auto testWithGivenBlockSize =
+      [](const IdTable& triplesInIndex, const IdTable& triplesToLocate,
+         size_t blockSizeInBytes,
+         const ad_utility::HashMap<size_t, std::string>&
+             expectedLocatedTriplesPerBlock) {
+        std::string basename = "LocatedTriplesTest.scanWithMergeTriples";
+        std::string permutationFilename = basename + ".index.pso";
+
+        // Collect the distinct relation `Id`s.
+        std::vector<Id> relationIds;
+        for (size_t i = 0; i < triplesInIndex.numRows(); ++i) {
+          relationIds.push_back(triplesInIndex(i, 0));
+          ASSERT_TRUE(i == 0 || relationIds[i - 1] <= relationIds[i]);
+        }
+        relationIds = ad_utility::removeDuplicates(relationIds);
+
+        // Helper lambda for creating a `BufferedIdTable` from all triples in
+        // the given `IdTable` matching `relationId`.
+        //
+        // This is needed need for `CompressedRelationWriter` below, which
+        // expects a `BufferedIdTable` with two columns.
+        //
+        // TODO: Something like this is also used in `CompressedRelationsTest`,
+        // so it should be in a helper class.
+        auto getBufferedIdTable = [](const IdTable& idTable,
+                                     Id relationId) -> BufferedIdTable {
+          // Note that these files are never created because we set the
+          // threshold for writing to disk so large.
+          std::string bufferFilename1 = "compressedRelationWriter.buffer1.dat";
+          std::string bufferFilename2 = "compressedRelationWriter.buffer2.dat";
+          AD_CONTRACT_CHECK(idTable.numColumns() == 3);
+          BufferedIdTable bufferedIdTable{
+              2, std::array{
+                     ad_utility::BufferedVector<Id>{
+                         std::numeric_limits<size_t>::max(), bufferFilename1},
+                     ad_utility::BufferedVector<Id>{
+                         std::numeric_limits<size_t>::max(), bufferFilename2}}};
+          for (size_t i = 0; i < idTable.size(); ++i) {
+            if (idTable(i, 0) == relationId) {
+              bufferedIdTable.push_back({idTable(i, 1), idTable(i, 2)});
+            }
+          }
+          return bufferedIdTable;
```

Comment:
I am currently working on a refactoring of the `Permutation` class, then it will become easier to add tthis functionality directtly to the `Permutation` class. 
So I think we're leaving this as is (I don't want to cause you more work than necessary) and throw it out, once we have the "proper" functionality for this).